File: dotnet/src/Model/Neutral/Units/Soldier.cs
using System;
using System.Collections.Generic;

namespace PlanetCS1 {
    public abstract class Soldier : Unit {
        public Soldier() {
            this.occ = true;
            this.traverse = false;
            this.minCost = 20;
            this.gasCost = 10;
            this.nrgCost = 0;
            this.health = 50;
            this.commands = new List<String>(){"Attack enemy"};
        }
    }
}

File: kotlinjs/old/Soldier.kt


abstract class Soldier() : Unit() {
    override var occ = true
    override var traverse = false
    override var minCost = 20
    override var gasCost = 10
    override var nrgCost = 0
    override var health = 50
    override var commands = mutableListOf<String>("Attack enemy")
}